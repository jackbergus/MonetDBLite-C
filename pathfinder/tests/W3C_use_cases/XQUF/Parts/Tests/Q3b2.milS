module(ascii_io);
module(unix);
module(pathfinder);
var TSTSRCDIR := getenv("TSTSRCDIR");
shred_doc(TSTSRCDIR+"/part-list.xml", "part-list.xml", "Parts", 9LL);
shred_doc(TSTSRCDIR+"/part-tree.xml", "part-tree.xml", "Parts", 9LL);

xquery("xml","
declare updating function
             local:delete-subtree($p as element(part))
  {
      for $child in doc(\"part-list.xml\")//part
      where $p/@partid eq $child/@partof
      return local:delete-subtree($child)
      ,
      do delete $p
  "+"};"+"

for $p in doc(\"part-tree.xml\")//part[@name=\"car\"]/part
return
  local:delete-subtree($p)
").printf();
fflush(stdout());
fflush(stderr());

xquery("xml","
doc(\"part-list.xml\")
").printf();
fflush(stdout());
fflush(stderr());

xquery("xml","
doc(\"part-tree.xml\")
").printf();
fflush(stdout());
fflush(stderr());

delete_doc("part-list.xml");
delete_doc("part-tree.xml");
