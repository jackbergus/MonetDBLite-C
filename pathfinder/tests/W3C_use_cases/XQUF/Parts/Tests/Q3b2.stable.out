stdout of test 'Q3b2` in directory 'tests/W3C_use_cases/XQUF/Parts` itself:


# 15:32:44 >  
# 15:32:44 >  Mtimeout -timeout 60 Mserver "--config=/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/etc/MonetDB.conf" --debug=10 --set "monet_mod_path=/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./PATHFINDER/lib/MonetDB:/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./PATHFINDER/lib/MonetDB/lib:/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/lib/MonetDB:/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/lib/MonetDB/lib:/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/lib/MonetDB/bin" --set "gdk_dbfarm=/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/var/MonetDB/dbfarm" --set "sql_logdir=/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/var/MonetDB/sql_logs" --set "sql_logdir=/Win/MonetDB/.GNU.32.32.d.--disable-optimize_--enable-debug.PREFIX./MONETDB/var/MonetDB/xquery_logs" --set mapi_port=39499 --set monet_prompt= --trace --dbname=mTests_tests_W3C_use_cases_XQUF_Parts  < Q3b2.milS
# 15:32:44 >  

# Monet Database Server V4.13.1
# Copyright (c) 1993-2006, CWI. All rights reserved.
# Compiled for i686-redhat-linux-gnu/32bit with 32bit OIDs; dynamically linked.
# Visit http://monetdb.cwi.nl/ for further information.
module(ascii_io);
module(unix);
module(pathfinder);
var TSTSRCDIR := getenv("TSTSRCDIR");
shred_doc(TSTSRCDIR+"/part-list.xml", "part-list.xml", "Parts", 9LL);
# shredding with holes: pagesize=65536, #fillsize=59638
# Elapsed time = 884us [036us/node]
# Shredded 1 XML document (part-list.xml), total time after commit=0.010s
shred_doc(TSTSRCDIR+"/part-tree.xml", "part-tree.xml", "Parts", 9LL);
# shredding with holes: pagesize=65536, #fillsize=59638
# Elapsed time = 386us [014us/node]
# Shredded 1 XML document (part-tree.xml), total time after commit=0.012s

xquery("xml","
declare updating function
             local:delete-subtree($p as element(part))
  {
      for $child in doc(\"part-list.xml\")//part
      where $p/@partid eq $child/@partof
      return local:delete-subtree($child)
      ,
      do delete $p
  "+"};"+"

for $p in doc(\"part-tree.xml\")//part[@name=\"car\"]/part
return
  local:delete-subtree($p)
").printf();

xquery("xml","
doc(\"part-list.xml\")
").printf();
fflush(stdout());
<?xml version="1.0" encoding="utf-8"?>
<XQueryResult>
<partlist>
  <part partid="0" name="car"/>
  <part partid="10" name="skateboard"/>
  <part partid="11" partof="10" name="board"/>
  <part partid="12" partof="10" name="wheel"/>
  <part partid="20" name="canoe"/>
</partlist>
</XQueryResult>

xquery("xml","
doc(\"part-tree.xml\")
").printf();
fflush(stdout());
<?xml version="1.0" encoding="utf-8"?>
<XQueryResult><parttree>
    <part name="car" partid="0">
        <part name="engine" partid="1">
            <part name="piston" partid="3"/>
        </part>
        <part name="door" partid="2">
            <part name="window" partid="4"/>
            <part name="lock" partid="5"/>
        </part>
    </part>
    <part name="skateboard" partid="10">
        <part name="board" partid="11"/>
        <part name="wheel" partid="12"/>
    </part>
    <part name="canoe" partid="20"/>
</parttree></XQueryResult>

delete_doc("part-list.xml");
delete_doc("part-tree.xml");
quit();

# 15:32:44 >  
# 15:32:44 >  Done.
# 15:32:44 >  

